### ISSUE TITLE
Adding a large amount of data to the M2M field will cause a lag

### ISSUE THREAD

**Issue by Loney-Cao** [odoo/odoo#50838]
Impacted versions:
**Odoo 13**
Steps to reproduce:

**Precondition**：
``` python
class ModelA(models.Model):
    _name = "model.a"
    ...
    fieldB =  fields.Many2many('model.b')
```
**Operating**
1. In the Form view of Model A, add a record to Field 'fieldB'。  
2.Field 'fieldB' has 10000 records。  
3.Add 2 more records to field 'fieldB' 。  
4.10002 records will be displayed on the current page

Current behavior:
**Browser rendering is very slow**

**Code path：**
`odoo/addons/web/static/src/js/views/basic/basic_model.js`
line：1831

``` js
_applyX2ManyChange: function (record, fieldName, command, options) {
                ......

                // Ensure the local data repository (list) boundaries can handle incoming records (data)
                if (data.length + list.res_ids.length > list.limit) {
                    list.limit = data.length + list.res_ids.length;    //   👈This method should not be used for large amounts of data
                }
```

Expected behavior:

Video/Screenshot link (optional):

**Comment by pedrobaeza** [comment]
I don't get the steps you are specifying for reproducing the problem.

**Comment by Loney-Cao** [comment]
> I don't get the steps you are specifying for reproducing the problem.

Maybe my English is not good, but you can try adding 10000 records for M2M type fields in the form view。Browser rendering is very slow

**Comment by pedrobaeza** [comment]
Yeah, and that's something expected, but pagination should work on this case. Can you put an screenshot of what you are trying to describe?

**Comment by Loney-Cao** [comment]
> Yeah, and that's something expected, but pagination should work on this case. Can you put an screenshot of what you are trying to describe?

OK,  this link is the screen record about the issue -->   [https://webmshare.com/play/WxGao ](https://webmshare.com/play/WxGao )

**Comment by pedrobaeza** [comment]
OK, thanks, now I understand, but I think that's an abuse of the existing tools, as it's a very extreme case. If you can provide a patch that fixed this, then it can be considered by Odoo team. Otherwise, you have to use an alternative way to handle this.

**Comment by Yenthe666** [comment]
@pedrobaeza I wouldn't really call it an abuse because there are some cases where you might need a lot of records/data. But yeah I'd recommend an alternative way to handle or show this data :) Let's call it a known limitation.

**Comment by pedrobaeza** [comment]
10K records attached to a m2m is not a normal use of this field IMO, but well, known limitation concept can fit also. The question is that this will be hardly solved without an initiative from the affected part.

**Comment by Yenthe666** [comment]
I don't see a direct solution or fix coming for this honestly, for sure not in stable. Perhaps one of the core developers from Odoo can give a final opinion on this? @mart-e and @xmo-odoo and @Elkasitu can I ask from input from one of you please?

**Comment by pedrobaeza** [comment]
This is more something of the frontend team I would say...

**Comment by Yenthe666** [comment]
Hmm you might be right.. @ged-odoo is our guy then

**Comment by Loney-Cao** [comment]
> Hmm you might be right.. @ged-odoo is our guy then

**Thank you for your attention！**
We have many scenes about a large amount of data, such as the factory has 5K employees, or more document information.
My solution is to delete code `list.limit = data.length + list.res_ids.length;`  And you can also insert the newly added record at the end of the current page record instead of the end of all records

**Comment by ruter** [comment]
For reference only, a solution to solve this issue, just remove these lines of code, but I haven't check is there other relevant potential risks.

The main problem is that the `list.limit` was changed and then it'll trigger data read from the field.

https://github.com/odoo/odoo/blob/3c55cae48936e195ec3411c55f577422c56828c3/addons/web/static/src/js/views/basic/basic_model.js#L1830-L1833

**Comment by pedrobaeza** [comment]
Probably that is going to have side effects.

**Comment by Loney-Cao** [comment]
@Yenthe666  @pedrobaeza  
On which page the record is added, the record is displayed on that page
Video address ：[https://showmore.com/zh/u/hyoi1s5](https://showmore.com/zh/u/hyoi1s5)

**Comment by pedrobaeza** [comment]
I don't know, but if you have an enterprise contract, you should raise an enterprise ticket on odoo.com/help for this to be analyzed. If not, you don't have any guarantee that this will be attended.

**Comment by jlzhou** [comment]
> For reference only, a solution to solve this issue, just remove these lines of code, but I haven't check is there other relevant potential risks.
> 
> The main problem is that the `list.limit` was changed and then it'll trigger data read from the field.
> 
> https://github.com/odoo/odoo/blob/3c55cae48936e195ec3411c55f577422c56828c3/addons/web/static/src/js/views/basic/basic_model.js#L1830-L1833

This fix broke limit setting in tree view:

https://github.com/odoo/odoo/commit/61216f116ac5c6e144c693b15408ad13db9f7ed4

**Comment by pivi-odoo** [comment]
Thank you for reporting this performance issue.

As part of our effort to clean up and triage performance-related issues, and because this issue concerns an older version of Odoo, we are closing it.

If you still experience the same problem on a currently supported version, please open a new issue with updated reproduction steps, relevant logs, and performance measurements.
Or open a ticket on odoo.com/help if you have an ongoing subscription for faster assistance.

Thank you for your understanding.
